### DISCUSSION TITLE
We are getting errors in airflow when we try to run PythonVirtualenvOperator tasks

### DISCUSSION THREAD

**Discussion by Jairam86** [apache/airflow#40570] [Q&A]
Airflow is installing different versions of pip installer for different task attempts and also the virtual env is getting built differently for each attempt which is causing the DAG to fail for some attempts and then succeed only on the reruns. 
[dag_id=poa-de-paa-hrp-prepay-batch-hrp-prepay-batch-239_run_id=scheduled__2024-06-18T14_30_00 00_00_task_id=send_to_bot_piab_py_attempt=7.log](https://github.com/user-attachments/files/16075994/dag_id.poa-de-paa-hrp-prepay-batch-hrp-prepay-batch-239_run_id.scheduled__2024-06-18T14_30_00.00_00_task_id.send_to_bot_piab_py_attempt.7.log)
[dag_id=poa-de-paa-hrp-prepay-batch-hrp-prepay-batch-239_run_id=scheduled__2024-06-18T14_30_00+00_00_task_id=send_to_bot_piab_py_attempt=4.log](https://github.com/user-attachments/files/16075995/dag_id.poa-de-paa-hrp-prepay-batch-hrp-prepay-batch-239_run_id.scheduled__2024-06-18T14_30_00%2B00_00_task_id.send_to_bot_piab_py_attempt.4.log)

**Comment by potiuk** [comment]
Sure. You should pin dependencies you need in your requirements.txt file if you want reproducibility. This is standard expectation any time you use requirement file - pinning dependencies you want to have installed in your venv is on you

**Reply by Jairam86** [reply]
Hi potuik,

Do you mean we need to set the pip version also in the requirements.txt file

Thanks,
Jairam
